### ISSUE TITLE
How to configure java_home being used?

### ISSUE THREAD

**Issue by corneliusroemer** [epi2me-labs/labslauncher#34]
I'm trying to run wf-artic on my ARM Mac.

I have openjdk 19.0.1 homebrew installed and linked as the system java version.

Unfortunately nextflow seems to require java between 8 and 18.

So when I run wf-artic I get the following error:
```
ERROR: Cannot find Java or it's a wrong version -- please make sure that Java 8 or later (up to 18) is installed

NOTE: Nextflow is trying to use the Java VM defined by the following environment variables:

JAVA_CMD: /opt/homebrew/Cellar/openjdk/19.0.1/libexec/openjdk.jdk/Contents/Home/bin/java

JAVA_HOME:
```

I'm not sure how to set that environment variable - it would be nice if this could be done within labslauncher so as to limit the impact on the rest of the system.

Any tips?

This is the config of the run:
```
{
"wfversion": "v0.3.18",
"validate_params": true,
"out_dir": "/Users/corneliusromer/epi2melabs-data/nextflow/instances/2022-11-29-00-26_wf-artic_Jezt5dFLvKkstyYMBfUQEv/output",
"scheme_name": "SARS-CoV-2",
"scheme_version": "Midnight-ONT/v3",
"report_name": "report",
"list_schemes": false,
"update_data": false,
"pangolin_version": "4.1.1",
"nextclade_version": "1.11.0",
"normalise": 200,
"medaka_model": "r941_min_hac_variant_g507",
"report_depth": 100,
"report_clade": true,
"report_coverage": true,
"report_lineage": true,
"report_variant_summary": true,
"report_detailed": false,
"fastq": "/Volumes/Macintosh HD/Users/corneliusromer/Downloads/BJGMC_ONT_Run71/barcode92.fastq.gz",
"wf": {
"agent": "labslauncher/3.1.5"
}
}
```

**Comment by claczny** [comment]
A bit lengthy text follows to provide context and the solution that worked for me is described at the end.

I haven't installed `labslauncher`yet, but was facing issues when launching the EPI2MELabs.app on a MacBook Pro M1 Pro with Ventura 13.1 installed.

I installed:
- `docker`via Docker-Desktop.
-  `nextflow` as described in the `nextflow` instructions.
- Java via `homebrew`, i.e., the `temurin` package. I first had Java-19 installed, which did *not* work with `nextflow`as `nextflow`complained and required/requires <= Java-18. I then installed `temurin-18` but it still tried to access the v.19.
This was solved by specifying
```
# In order to use Nextflow, <=Java-18 is needed. In the future, maybe the `-v 18` can be removed to ensure that the very latest java version is used. For now (2023-01-06), it does not matter.
# Easy version switching found at https://gist.github.com/tinkerware/8d92524d78f958f3d821b127393a96a1?permalink_comment_id=2716292#gistcomment-2716292
export JAVA_HOME=$(/usr/libexec/java_home -v 18)
```

I thought this was then all set but when launching the EPI2MELabs.app from the Finder by clicking on it, it still said that there were some issue with Java and nextflow.
I was reluctant to let the app do the installation of Java and `nextflow` for two reasons:
1. Java and `nextflow` are nicely installable using other means, so why have another app handle this. This is however a nice feature for uninitiated people, so kudos to the devs!
2. Letting the app install them would only minorly bloat the disk, so it would be fine, BUT, it was not clear from the docs where Java and `nextflow` from the app would be installed and if this would interfere with my regular installs. So this could maybe be added to the Installation docs as a clarification.

However, and here comes the helpful part, if I launch the app via the terminal, it works fine :)
So it nicely takes the set-up from the shell environment and all is well!

**tl;dr**
Hence, maybe you can try that too? I.e.

- install `temurin18`, either by installing it in addition to the `temurin`package or uninstall `temurin`
- in case of the former, use `export JAVA_HOME=$(/usr/libexec/java_home -v 18)` in your shell config (e.g., in `$HOME/.zshrc`) to ensure the right version is used or just have it in front of the `launcherlabs` call

My config is:
```
cedric.laczny:~/ $ java --version                                                                                              [15:45:19]
openjdk 18.0.2.1 2022-08-18
OpenJDK Runtime Environment Temurin-18.0.2.1+1 (build 18.0.2.1+1)
OpenJDK 64-Bit Server VM Temurin-18.0.2.1+1 (build 18.0.2.1+1, mixed mode)
cedric.laczny:~/ $ nextflow -v                                                                                                 [15:45:26]
nextflow version 22.10.4.5836
```
